# Parameters for ROSflight software-in-the-loop simulation.
# Mass and inertia parameters are defined in multirotor.urdf.xacro.

/**:
  ros__parameters:
    # Common Global Physical Parameters
    rho: 1.225

    # Dynamics
    num_rotors: 4
    rotor_positions: [ 0.1926,  0.230, -0.0762,
                      -0.1907,  0.205, -0.0762,
                      -0.1907, -0.205, -0.0762,
                       0.1926, -0.230, -0.0762 ]
    
    rotor_dists: [.25, .25, .25, .25]
    rotor_radial_angles: [45., 135., 225., 315.]
    rotor_rotation_directions: [ -1, 1, -1, 1 ]
    
    # Prop Model
    # Thrust
    CT_0: 0.08729101
    CT_1: -0.11006411
    CT_2: -0.12928596

    # Torque
    CQ_0: 0.00428594
    CQ_1: 0.00146105
    CQ_2: -0.01440988

    # Diameter (m)
    D_prop: 0.381
    
    # Motor Specs
    V_max : 24.0                  # voltage for 6s battery at 4 volts per cell
    R_motor : 0.042               # ohms
    I_0 : 1.5                     # no-load (zero torque) current (A)
    KQ : 0.028937                 #((1 /(KV_rpm_per_volt)) *60) / (2 *M_PI) Back-emf constant, KV in V-s/rad, Motor torque constant, KQ in N-m/A

    # Drag Model
    CD: 2.05                      # Coeffecient of drag for a rough cube face on into flow.
    A_c: 0.04                     # Crossectional Area to use in Drag.

    # Induced Drag Model
    CD_induced: 0.05

    # Ground Effect Model
    ground_effect: [ -55.3516, 181.8265, -203.9874, 85.3735, -7.6619 ]
